### ISSUE TITLE
Property 'scrollToEnd' does not exist on type 'SwipeableFlatList<any>'.ts(2339)

### ISSUE THREAD

**Issue by jkoutavas** [esthor/react-native-swipeable-list#69]
Can someone show me an example of how to setup the ref on SwipeableFlatlist in order to call scrollToEnd() on it?

If I setup my ref as such:

```
const flatlistRef = useRef<SwipeableFlatList>(null);
```

I get this tsc error:

<img width="926" alt="image" src="https://github.com/esthor/react-native-swipeable-list/assets/1264625/c9103d8b-f5ba-4c3f-9ced-e26221d7c348">

**Comment by esthor** [comment]
Hey @jkoutavas! I think it's because this library isn't currently exporting TypeScript type definitions. Does it work functionally and just complain about the type?

Basically, I think your setup and usage could be fine, but TypeScript will throw this error (likely) because your TypeScript compiler is saying "Hey! What a minute...I'm inferring an `any` type on anything that I don't have a type definition or can infer from the code, and specific things like `scrollToEnd()` isn't part of the `any` type."

Does that make sense?

I've really been meaning to add those type definitions. I was hoping to move this community PR along (https://github.com/esthor/react-native-swipeable-list/pull/61), which would likely solve this for your case. If you'd like to reopen it and address the issues to get it ready, I'd greatly appreciate it! Otherwise, I can try to get around to it 🙏

**Comment by jkoutavas** [comment]
@esthor 

Thank you for the reply.

Were it just a typing issue, I wouldn't have minded. But, the error is deeper than that...

```
 ERROR  TypeError: _flatlistRef$current.scrollToEnd is not a function (it is undefined), js engine: hermes

**Comment by esthor** [comment]
Oh. I see. That's not great.

If you use FlatList from react-native, it works fine?

I have not tested on the latest version of React Native yet, so I'm wondering if that's going to be an issue. 🤔 😅

**Comment by jkoutavas** [comment]
Yeah, it works fine with FlatList. I'm trying to move my current implementation from it to SwipeableFlatList.

**Comment by jkoutavas** [comment]
FWIW, PR #70 doesn't fix this.

**Comment by esthor** [comment]
Ok, I may need to update this library to specifically use the latest FlatList version. Starting to test now...

**Comment by esthor** [comment]
Ok, I have a repro 😄

**Comment by esthor** [comment]
TL;DR -- adding refs on refs is instantly janky 😭 

Ok, so I've given it the old college try. And I'm at a point where it might make sense to refactor the library in order to support refs on top of it in a performant way.

What I've done:
- I've refactored the library to TS (to make everyone's life easier, including mine 😅 )
- I've updated the example app
- I've gotten refs to the swipeable flatlist to work, but the app just tanks in terms of performance.

I am at the point where I need to step back and think a bit more about what's the best path forward. Right now, I don't think there is a quick way to get refs to work for you that I'm seeing.

I'll come back in a week or so to try again. 😕

**Comment by jkoutavas** [comment]
@esthor I applaud your PR #74 efforts!

> I've gotten refs to the swipeable flatlist to work, but the app just tanks in terms of performance.

Where is this work, or is it too tacky to commit? :-)

**Comment by jkoutavas** [comment]
> TL;DR -- adding refs on refs is instantly janky 

Yeah. Is there anyway to expose the underlying FlatList ref as a property to SwipeableFlatList? Or am I being naive?

**Comment by esthor** [comment]
> Where is this work, or is it too tacky to commit? :-)

Lol, yeah, pretty tacky to commit right now. I'm going to have another go at it.

> Is there an approach to expose the underlying FlatList ref as a property to SwipeableFlatList?

Yeah, that's likely how I'll need to refactor how I was doing it. Larger context was that I was in the middle of a complete JS/FlowTypes -> TypeScript and Class->Function component rewrite when doing this. 😁 So there was a lot of change not fully tested at the time.

"Soon" 🙏

**Comment by jkoutavas** [comment]
@esthor I'm very appreciative that you are actively working on this. This package has been my go-to (pun intended) for a number of projects I've done. Let me know if there's anything more I can do to help.

**Comment by jkoutavas** [comment]
@esthor Happy Holidays. Any news to share?
